integrations/streamer_test.go

```diff
@@ -9027,6 +9027,178 @@ stream
+				Columns: []string{"time", "expiration"},
+				Values: [][]interface{}{
+					{
+						time.Date(1971, 1, 1, 0, 0, 0, 0, time.UTC),
+						float64(6.56419e+17), // 1990-10-20 10:42:42
+					},
+				},
+			},
+		},
+	}
+
+	testStreamerWithOutput(t, "TestStream_LambdaNow", script, time.Second, expectedOutput, false, nil)
+}
+
+func TestStream_EvalNow(t *testing.T) {
+	var script = `
+stream
+	|from().measurement('account')
+	|eval(lambda: now())
+	  .as('now')
+	|httpOut('TestStream_EvalNow')
+`
+
+	expectedOutput := models.Result{
+		Series: models.Rows{
+			{
+				Name:    "account",
+				Tags:    map[string]string{"owner": "ownerA"},
+				Columns: []string{"time", "expiration"},
+				Values: [][]interface{}{
+					{
+						time.Date(1971, 1, 1, 0, 0, 0, 0, time.UTC),
+						float64(3.15533e+17), // 1980-01-01 00:00:00 (Unix ns timestamp)
+
+						// we expect "expiration" to be float64 and not int64 (even with input data consisting of ints)
+						// because httpOut uses JSON as serialization format for the results,
+						// resulting in the integers becoming floats
+					},
+				},
+			},
+
+			{
+				Name:    "account",
+				Tags:    map[string]string{"owner": "ownerC"},
+				Columns: []string{"time", "expiration"},
+				Values: [][]interface{}{
+					{
+						time.Date(1971, 1, 1, 0, 0, 0, 0, time.UTC),
+						float64(6.56419e+17), // 1990-10-20 10:42:42
+					},
+				},
+			},
+		},
+	}
+
+	testStreamerWithOutput(t, "TestStream_EvalNow", script, time.Second, expectedOutput, false, nil)
+}
+
+func TestStream_EvalNowInfluxDBOut(t *testing.T) {
+	var script = `
+stream
+	|from()
+		.measurement('account')
+	|eval(lambda: now())
+		.as('now')
+	|influxDBOut()
+		.database('db')
+		.retentionPolicy('rp')
+		.measurement('m')
+		.precision('s')
+		.tag('key', 'value')
+		.flushInterval(1ms)
+`
+	done := make(chan error, 1)
+	var points []imodels.Point
+	var database string
+	var rp string
+	var precision string
+
+	influxdb := NewMockInfluxDBService(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/ping" {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+		//Respond
+		var data client.Response
+		w.WriteHeader(http.StatusOK)
+		_ = json.NewEncoder(w).Encode(data)
+		//Get request data
+		database = r.URL.Query().Get("db")
+		rp = r.URL.Query().Get("rp")
+		precision = r.URL.Query().Get("precision")
+
+		b, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			done <- err
+			return
+		}
+		points, err = imodels.ParsePointsWithPrecision(b, time.Unix(0, 0), precision)
+		done <- err
+	}))
+
+	tmInit := func(tm *kapacitor.TaskMaster) {
+		tm.InfluxDBService = influxdb
+	}
+	testStreamerNoOutput(t, "TestStream_EvalNowInfluxDBOut", script, 15*time.Second, tmInit)
+
+	if database != "db" {
+		t.Errorf("got %v exp %v", database, "db")
+	}
+	if rp != "rp" {
+		t.Errorf("got %v exp %v", rp, "rp")
+	}
+	if precision != "s" {
+		t.Errorf("got %v exp %v", precision, "s")
+	}
```

Comment:
I think we can disregard these checks here since they're not what we're really testing.

Reply:
Hi. I made the changes and added a lot of logging (nothing committed) to try to understand what was going on, since I'm not familiar with how kapacitor is built. It looks like there is no HTTP request made at all to InfluxDB. And it seems that eval is not even called (tried to log inside NewEvalLambdaNode for testing purposes and got nothing). I'll continue my investigations another day.